frmts/hdf5/hdf5dataset.cpp

```diff
@@ -1132,6 +1178,28 @@ static herr_t HDF5AttrIterate(hid_t hH5ObjID, const char *pszAttrName,
                 psContext->m_osValue += szData;
             }
         }
+#ifdef HDF5_HAVE_FLOAT16
+        else if (H5Tequal(H5T_NATIVE_FLOAT16, hAttrNativeType) > 0)
+        {
+            for (hsize_t i = 0; i < nAttrElmts; i++)
+            {
+                const uint16_t nVal16 = static_cast<uint16_t *>(buf)[i];
+                const uint32_t nVal32 = CPLHalfToFloat(nVal16);
```

Comment:
I'm confused by CPLHalfToFloat (which doesn't return a float) and the memcpy below (which assumes sizeof float), I know that they are both 32 bits but I am wondering if that assumption is enforced by standard or if is it "assert"ed somewhere.

Reply:
We have a ``static_assert(sizeof(float) == 4);`` in port/cpl_conv.cpp